### PR TITLE
[macos] Add explicit 57 error instead of NullReferenceException when …

### PR DESCRIPTION
**Pull request by chamons**
…building without --sdkroot set

### CODE REVIEW

**Comment by chamons** [comment]
Ran across this while compiling mac-samples with master

**Comment by chamons** [comment]
So i'm using the same error, with '' instead of the path (since it was not set).

I can make a new error code instead of reusing 57, but i'm not sure what to update if I use the same message?

**Comment by spouliot** [comment]
it's not about re-using the number - it's because `docs/website/mmp-errors.md` does not cover `57` (like mtouch's version). IOW if you add a new code in mmp then `mmp-errors.md` must be updated too :)

**Comment by chamons** [comment]
So the reason that I was confused is that driver.cs is _already_ using 57:

https://github.com/xamarin/xamarin-macios/blob/master/tools/mmp/driver.cs#L565

But yeah, it isn't documented. Let me add that.

**Comment by chamons** [comment]
Both my PRs I built hit the same issue. Let's try building one more time to make sure it wasn't network...

**Comment by chamons** [comment]
build

**Comment by chamons** [comment]
That seems reasonable @rolfbjarne . I am all for XM and XI having the same behavior when possible.

Let me do that instead...

File: tools/mmp/driver.cs [hunk 1]
```diff
@@ -551,9 +551,19 @@ static bool ParseBool (string value, string name)
...
 
 
+		const string XcodeDefault = "/Applications/Xcode.app";
 		static void ValidateXcode ()
 		{
 			if (xcode_version == null) {
+				if (sdk_root == null) {
+					if (File.Exists (Path.Combine (XcodeDefault, "Contents/MacOS/Xcode"))) {
+						sdk_root = XcodeDefault;
+						ErrorHelper.Warning (62, "No Xcode.app specified (using --sdkroot or 'xcode-select --print-path'), using the default Xcode instead: {0}", sdk_root);
```
Comment (rolfbjarne) [inline tools/mmp/driver.cs]: You're not actually checking the system here (even though the warning message says so). Can't we just move this entire chunk of code to the shared Driver.cs and fixup iOS specific stuff: https://github.com/xamarin/xamarin-macios/blob/363eb08/tools/mtouch/mtouch.cs#L319-L397?

**Comment by chamons** [comment]
Let's re-do this later, since Rolf is correct but that will involve more work that I want to bite off right now.

https://trello.com/c/sD9xu1MG/324-rewrite-xcode-detection-logic-in-mmp

**Pull request closed without merging**